EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -1,1745 +1,1220 @@
+	{
+		importPopsFromFile(filename, minorityPopMapper);
+	}
+}
+
+void V2::World::importPopsFromFile(const std::string& filename, const mappers::MinorityPopMapper& minorityPopMapper)
+{
+	std::list<int> popProvinces;
+
+	// We are using our own defaults instead of vanilla source because it was modded with cultural minorities.
+	const mappers::PopMapper popMapper("./blankMod/output/history/pops/1836.1.1/" + filename);
+
+	for (const auto& provinceItr : popMapper.getProvincePopTypeMap())
+	{
+		int provinceNum = provinceItr.first;
+		popProvinces.push_back(provinceNum);
+
+		importPopsFromProvince(provinceNum, provinceItr.second, minorityPopMapper);
+	}
+	popRegions.insert(std::make_pair(filename, popProvinces));
+}
+
+void V2::World::importPopsFromProvince(int provinceID, const mappers::PopTypes& popType, const mappers::MinorityPopMapper& minorityPopMapper)
+{
+	int provincePopulation = 0;
+	int provinceSlavePopulation = 0;
+
+	auto province = provinces.find(provinceID);
+	if (province == provinces.end())
+	{
+		LOG(LogLevel::Warning) << "Could not find province " << provinceID << " for original pops.";
+		return;
+	}
+
+	for (const auto& pop: popType.getPopTypes())
+	{
+		auto newPop = std::make_shared<Pop>(pop.first, pop.second.getSize(), pop.second.getCulture(), pop.second.getReligion());
+		if (minorityPopMapper.blankMajorityFromMinority(*newPop))
+		{
+			// If the pop we loaded had minority elements, their majority elements are now blank.
+			// Ie. We've removed yankee from mormons, and we'll fill them later dynamically to become <majorityculture> mormons.
+			province->second->addMinorityPop(newPop);
+		}
+		province->second->addVanillaPop(newPop);
+
+		totalWorldPopulation += newPop->getSize();
+		provincePopulation += newPop->getSize();
+		if (newPop->isSlavePop()) provinceSlavePopulation += newPop->getSize();
+	}
+
+	province->second->setSlaveProportion(1.0 * provinceSlavePopulation / provincePopulation);
+}
+
+void V2::World::importPotentialCountries()
+{
+	potentialCountries.clear();
+	dynamicCountries.clear();
+
+	std::ifstream V2CountriesInput;
+	V2CountriesInput.open("./blankMod/output/common/countries.txt");
+	if (!V2CountriesInput.is_open())
+	{
+		LOG(LogLevel::Error) << "Could not open countries.txt. The converter may be corrupted, try downloading it again.";
+		exit(-1);
+	}
+
+	bool dynamicSection = false;
+	while (!V2CountriesInput.eof())
+	{
+		std::string line;
+		getline(V2CountriesInput, line);
+
+		if ((line[0] == '#') || (line.size() < 3))
+		{
+			continue;
+		}
+		else if (line.substr(0, 12) == "dynamic_tags")
+		{
+			dynamicSection = true;
+			continue;
+		}
+
+		importPotentialCountry(line, dynamicSection);
+	}
+
+	V2CountriesInput.close();
+}
+
+void V2::World::importPotentialCountry(const std::string& line, bool dynamicCountry)
+{
+	std::string tag = line.substr(0, 3);
+
+	auto newCountry = std::make_shared<Country>(line, dynamicCountry, partyNameMapper, partyTypeMapper);
+	potentialCou
```

Comment:
It might still be worth shoving in a named constant at the top of the file or somewhere: ```constexpr int MAX_EQUALITY_COUNTRIES = 5;```

Reply:
done